src/main/webapp/definitions/alfresco-core.yaml

```diff
@@ -10683,6 +10701,9 @@ definitions:
       attachment:
         type: boolean
         description: URL type (embedded/attachment).
+      fileName:
```

Comment:
I think we'd usually have the items from the "Create" entity in the response too - i.e. `DirectAccessUrl` below. I don't think this is explicitly called out in the v1 API guidelines though.

I've added our architect team to review too, since they usually review additions to the v1 API.

Reply:
I could add it to the `DirectAccessUrl` class in alfresco-community-repo, but it looks like the current object structure is fed in from the content store itself. Not exactly sure where that code is or how dangerous it is to alter.
I could make `DirectAccessUrlResponse` class for the purposes of extending an adding on the additional field.
It's not hard to return the field, I just want to keep things in line with your expectations for the code base as a whole.

Reply:
@tpage-alfresco is correct, please make sure to add the file name to the response too. 
In terms of implementation, you have already modified the `ContentServiceImpl` class, so **adding** an attribute to `DirectAccessUrl` (part of our Java public API) is not a problem.

Reply:
@jamalkm The issue is the call to the content store serialized directly into` DirectAccessUrl`. While the serialization might still work with an extra field added, I worry it might start to diverge from representing the data object actually returned from the database vs. what we are returning from the API.

```
DirectAccessUrl directAccessUrl = null;
if (store.isContentDirectUrlEnabled())
{
    try
    {
        // referencing this
        directAccessUrl = store.requestContentDirectUrl(contentUrl, attachment, fileName, contentMimetype, validFor);
    }
    catch (UnsupportedOperationException ex)
    {
        // expected exception
    }
}
return directAccessUrl;
```
I can still open a PR on it though

Reply:
https://github.com/Alfresco/alfresco-community-repo/pull/2081